### PR TITLE
Add recommened roles for Elastic Agent on Kubernetes

### LINKED ISSUE #8168: Elastic Agent Kubernetes Integration missing default permissions
We are using ECK with Elastic Agents Managed by Fleet and have installed the Kubernetes Integration.

We have noticed a number of errors relating to permissions;

The first two i believe can be resolved by updating the cluster role: https://github.com/elastic/cloud-on-k8s/blob/main/deploy/eck-stack/charts/eck-agent/values.yaml#L127-L183

> {"log.level":"error","@timestamp":"2024-10-31T10:18:59.704Z","message":"E1031 10:18:59.703948    1125 reflector.go:147] k8s.io/client-go@v0.29.5/tools/cache/reflector.go:229: Failed to watch *v1.DaemonSet: failed to list *v1.DaemonSet: daemonsets.apps is forbidden: User \"system:serviceaccount:elastic-system:elastic-agent\" cannot list resource \"daemonsets\" in API group \"apps\" at the cluster scope","component":{"binary":"metricbeat","dataset":"elastic_agent.metricbeat","id":"kubernetes/metrics-default","type":"kubernetes/metrics"},"log":{"source":"kubernetes/metrics-default"},"ecs.version":"1.6.0"}

> {"log.level":"error","@timestamp":"2024-10-31T09:26:58.580Z","message":"E1031 09:26:58.580250    1036 reflector.go:147] k8s.io/client-go@v0.29.5/tools/cache/reflector.go:229: Failed to watch *v1.PersistentVolume: failed to list *v1.PersistentVolume: persistentvolumes is forbidden: User \"system:serviceaccount:elastic-system:elastic-agent\" cannot list resource \"persistentvolumes\" in API group \"\" at the cluster scope","component":{"binary":"metricbeat","dataset":"elastic_agent.metricbeat","id":"kubernetes/metrics-default","type":"kubernetes/metrics"},"log":{"source":"kubernetes/metrics-default"},"ecs.version":"1.6.0"}

We are unsure why we are receiving the below 401 errors.

> {"log.level":"error","@timestamp":"2024-10-31T09:30:20.489Z","message":"add_cloud_metadata: received error failed with http status code 401","component":{"binary":"metricbeat","dataset":"elastic_agent.metricbeat","id":"http/metrics-monitoring","type":"http/metrics"},"log":{"source":"http/metrics-monitoring"},"service.name":"metricbeat","ecs.version":"1.6.0","log.logger":"add_cloud_metadata","log.origin":{"file.line":190,"file.name":"add_cloud_metadata/providers.go","function":"github.com/elastic/beats/v7/libbeat/processors/add_cloud_metadata.(*addCloudMetadata).fetchMetadata"},"ecs.version":"1.6.0"}

> {"log.level":"error","@timestamp":"2024-10-31T09:30:15.788Z","message":"add_cloud_metadata: received error failed with http status code 401","component":{"binary":"filebeat","dataset":"elastic_agent.filebeat","id":"filestream-monitoring","type":"filestream"},"log":{"source":"filestream-monitoring"},"log.origin":{"file.line":190,"file.name":"add_cloud_metadata/providers.go","function":"github.com/elastic/beats/v7/libbeat/processors/add_cloud_metadata.(*addCloudMetadata).fetchMetadata"},"service.name":"filebeat","ecs.version":"1.6.0","log.logger":"add_cloud_metadata","ecs.version":"1.6.0"}

### PR DESCRIPTION
**Pull request by pebrc**
Fixes #8168 

I tried to amend the roles based on https://raw.githubusercontent.com/elastic/elastic-agent/v8.15.3/deploy/kubernetes/elastic-agent-standalone-kubernetes.yaml 

I left out the cloudbeat recommendations, because I think we cannot cater for all integrations in the defaults. 
But we can also go for a the maximal set of permissions, to avoid problems and allow scurity conscious users to trim them down to what is necessary for them.

It is problematic that we need to update ~ 4 places and effectively duplicate the roles that the Elastic Agent team defines without having a good process to update them when the source of truth changes.

### CODE REVIEW

File: deploy/eck-stack/charts/eck-agent/values.yaml [hunk 1]
```diff
@@ -127,57 +127,71 @@ clusterRoleBinding:
...
+  verbs:
+  - get
+  - create
+  - update
+- nonResourceURLs:
+  - "/metrics"
+  verbs:
+  - get
+- apiGroups: ["extensions"]
+  resources:
```
Comment (barkbay) [inline deploy/eck-stack/charts/eck-agent/values.yaml:154]:
```suggestion
  - apiGroups: [""]
    resources:
    - pods
    - nodes
    - namespaces
    - events
    - services
    - configmaps
    verbs:
    - get
    - watch
    - list
  - apiGroups: ["coordination.k8s.io"]
    resources:
    - leases
    verbs:
    - get
    - create
    - update
  - nonResourceURLs:
    - "/metrics"
    verbs:
    - get
  - apiGroups: ["extensions"]
    resources:
```

File: deploy/eck-stack/charts/eck-agent/values.yaml [hunk 2]
```diff
@@ -127,57 +127,71 @@ clusterRoleBinding:
...
+  - "list"
+  - "watch"
+- apiGroups:  
+  - "storage.k8s.io" 
+  resources:
+  - storageclasses
+  verbs:  
+  - "get"
+  - "list"
+  - "watch"
```
Comment (barkbay) [inline deploy/eck-stack/charts/eck-agent/values.yaml:197]:
```suggestion
    verbs:
    - "get"
    - "list"
    - "watch"
  - apiGroups:
    - "apps"
    resources:
    - statefulsets
    - deployments
    - replicasets
    - daemonsets
    verbs:
    - "get"
    - "list"
    - "watch"
  - apiGroups:
    - ""
    resources:
    - nodes/stats
    verbs:
    - get
  - nonResourceURLs:
    - "/metrics"
    verbs:
    - get
  - apiGroups:
    - "batch"
    resources:
    - jobs
    - cronjobs
    verbs:
    - "get"
    - "list"
    - "watch"
  - apiGroups:
    - "storage.k8s.io"
    resources:
    - storageclasses
    verbs:
    - "get"
    - "list"
    - "watch"
```

**Review by barkbay — APPROVED** [review]
Indentation seems wrong in one file, LGTM otherwise.

**Comment by pebrc** [comment]
buildkite test this -f p=gke,s=8.16.0,E2E_TAGS=agent

**Comment by pebrc** [comment]
buildkite test this -f p=gke,s=8.16.0,E2E_TAGS=agent

**Comment by pebrc** [comment]
buildkite test this -f p=gke,s=8.16.0,E2E_TAGS=agent

**Pull request merged**